File: problem/Kotlin/kids-with-the-greatest-number-of-candies.kt
class Solution {
    fun kidsWithCandies(candies: IntArray, extraCandies: Int): BooleanArray {
        val lowInt:Int = (candies.max()?:0) - extraCandies
        val arRet = mutableListOf<Boolean>()
        for(num in candies)
            if(num >= lowInt)
                arRet.add(true)
            else
                arRet.add(false)
        return arRet.toBooleanArray()
    }
}

File: problem/C++/kids-with-the-greatest-number-of-candies.cpp
class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
        vector<bool> ans;
        int val = *max_element(candies.begin(), candies.end()) - extraCandies;
        
        for(auto i : candies)
            if(i >= val)
                ans.push_back(true);
            else
                ans.push_back(false);
        return ans;
    }
};